docs/java/how-to-define-a-query-in-java.md

```diff
@@ -0,0 +1,31 @@
+---
+id: how-to-define-a-query-in-java
+title: How to define a Query in Java
+sidebar_label: Define Query
+description: Create a Query handler using the `@QueryMethod` annotation in the Workflow interface.
+tags:
+  - java
+  - developer-guide
+---
+
+To define a Query, create a Query handler using the `@QueryMethod` annotation in the Workflow interface.
```

Comment:
What we need to do is show that Queries are referenced at both ends by a name. And the name is just a string.

But the other part is defining the data structure for the data that the Query will end up returning.

When defining a Query handler in java - the method name either needs to exactly match the Query name OR it needs an annotation to match the name.

The name is somewhat separate from the Query handler - 

When sending a Query - you are just using the AP and specifying a name and getting a structured response

Its likely that these will be defined as separate things and reused across multiple Workflow types.